src/cpr_data_access/parser_models.py

```diff
@@ -157,44 +166,121 @@ class ParserOutput(BaseModel):
 
     @root_validator
     def check_html_pdf_metadata(cls, values):
-        """Check that html_data is set if content_type is HTML, or pdf_data is set if content_type is PDF."""
+        """
+        Validate the relationship between content-type and the data that is set.
+
+        Check that html_data is set if content_type is HTML, or pdf_data is set if
+        content_type is PDF.
+
+        Check that if the content-type is not HTML or PDF, then html_data and pdf_data
+        are both null.
+        """
         if (
             values["document_content_type"] == CONTENT_TYPE_HTML
             and values["html_data"] is None
         ):
-            raise ValueError("html_metadata must be set for HTML documents")
+            raise ValueError("html_data must be set for HTML documents")
 
         if (
             values["document_content_type"] == CONTENT_TYPE_PDF
             and values["pdf_data"] is None
         ):
             raise ValueError("pdf_data must be set for PDF documents")
 
-        if values["document_content_type"] is None and (
-            values["html_data"] is not None or values["pdf_data"] is not None
-        ):
+        if values["document_content_type"] not in {
+            CONTENT_TYPE_HTML,
+            CONTENT_TYPE_PDF,
+        } and (values["html_data"] is not None or values["pdf_data"] is not None):
             raise ValueError(
-                "html_metadata and pdf_metadata must be null for documents with no content type."
+                "html_data and pdf_data must be null for documents with no content type."
             )
 
         return values
 
     @property
-    def text_blocks(self) -> Sequence[TextBlock]:  # type: ignore
+    def text_blocks(self) -> Sequence[TextBlock]:
         """
-        Return the text blocks in the document. These could differ in format depending on the content type.
+        Return the text blocks in the document.
+
+        These could differ in format depending on the content type.
 
         :return: Sequence[TextBlock]
         """
-
         if self.document_content_type == CONTENT_TYPE_HTML:
-            return self.html_data.text_blocks  # type: ignore
+            html_data: Union[HTMLData, None] = self.html_data
```

Comment:
`Optional[HTMLData]`? same with below

Reply:
Because we have `None` values applied if `HTMLData` is not supplied during object instantiation then we really only should expect `None` or `HTMLData`

`pdf_data: Optional[PDFData] = None`